modules/net/quic/socket.c

```diff
@@ -2385,6 +2416,68 @@ static int quic_sock_get_crypto_secret(struct sock *sk, u32 len,
 	return 0;
 }
 
+static int quic_sock_get_stream_peeloff(struct sock *sk, u32 len,
+					sockptr_t optval, sockptr_t optlen)
+{
+	struct quic_stream_table *streams = quic_streams(sk);
+	struct quic_stream_peeloff peeloff = {};
+	struct quic_frame *frame, *next;
+	struct quic_stream *stream;
+	struct socket *sock;
+	struct file *file;
+	int retval;
+
+	if (len < sizeof(peeloff) || !quic_is_established(sk))
+		return -EINVAL;
+	len = sizeof(peeloff);
+	if (copy_from_sockptr(&peeloff, optval, len))
+		return -EFAULT;
+
+	/* Do not peel off from one netns to another one. */
+	if (!net_eq(current->nsproxy->net_ns, sock_net(sk)))
+		return -EINVAL;
```

Comment:
Interesting. I would think that this would not be needed, as holding the "master socket" is enough privileges. But maybe there is some other interaction I am not aware of?

Sending the peeled off socket to a process in another net namespace is still fine, right?